### DISCUSSION TITLE
Using operationPreRequestScripts

### DISCUSSION THREAD

**Discussion by arabelladavey1** [apideck-libraries/portman#499] [General]
Hi, I have a large collection pre-request script with multiple collection variables and functions defined. I was wondering if it is possible to use operationPreRequestScripts to inject this pre-request script into numerous collections and if there is an example I could look at? Thanks

**Comment by thim81** [comment]
Portman supports passed pre-request script inline or from files:
https://github.com/apideck-libraries/portman#portman---operationprerequestscripts-properties

An example based on a [test](https://github.com/apideck-libraries/portman/blob/bba46b0d47a16d45bb3663886652bd7471c89cc4/src/application/tests/extendTest.test.ts#L77) but the concept is the same for [operationPreRequestScripts](https://github.com/apideck-libraries/portman/blob/bba46b0d47a16d45bb3663886652bd7471c89cc4/examples/cli-options/portman-config.crm.yaml#L19) 

As long as the `file:` is accessible for Portman you could centralize the pre-request scripts and re-use them in various projects.
